lib/rules/eqeqeq.js

```diff
@@ -79,11 +83,17 @@ module.exports = function(context) {
                 return;
             }
 
-            context.report(
-                node, getOperatorLocation(node),
-                "Expected '{{op}}=' and instead saw '{{op}}'.",
-                {op: node.operator}
-            );
+            context.report({
+                nodeType: node.type,
+                loc: getOperatorLocation(node),
+                message: "Expected '{{op}}=' and instead saw '{{op}}'.",
+                data: { op: node.operator },
+                fix: function(fixer) {
+                    return fixer.insertTextAfter(sourceCode.getTokenAfter(node.left).range, "=");
+
+                }
+            });
+
```

Comment:
New signature for `context.report()` allows you to specify a fix function to along with the message. The return value specifies the operation to perform, so you can only perform one operation per rule (greatly simplifies reconciliation later).